compiler/CodeGenVisitor.h

```diff
@@ -4,26 +4,51 @@
 #include "generated/ifccBaseVisitor.h"
 using namespace std;
 
+struct Element
```

Comment:
pls add comments to what this Element struct is and why we use it.

If you have time I think it would be nice to describe why we use Variable and Function as well

Reply:
I'll add the descriptions